### PR TITLE
feat: updates for doc rearchitecture, part 1

### PR DESCRIPTION
**Pull request by jcstein**
## Overview

This is the first of a few PRs to update docs architecture. It does the following:

| Change | Preview Link (good while PR is open) | Final Link (only good after PR is merged) |
| --- | --- | --- |
| Moves IBC docs to Nodes category | [Preview](https://docs.celestia.org/pr-preview/pr-1385/nodes/ibc-relayer) | [Final link](https://docs.celestia.org/nodesx/ibc-relayer) |
| Moves "Retrievability" page to Learn > Overview > Data availability layer menu | [Preview](https://celestiaorg.github.io/docs/pr-preview/pr-1385/learn/retrievability) | [Final Link](https://docs.celestia.org/learn/retrievability) |
| Moves staking docs, "How to stake TIA" and "Staking dashboards" to learn section | [Preview](https://celestiaorg.github.io/docs/pr-preview/pr-1385/learn/how-to-stake-tia) | [Final Link](https://docs.celestia.org/learn/how-to-stake-tia) |
| Moves "Celestia-app wallet" page to Node category | [Preview](https://celestiaorg.github.io/docs/pr-preview/pr-1385/nodes/celestia-app-wallet) | [Final Link](https://docs.celestia.org/nodes/celestia-app-wallet) |
| Moves Multisig doc to Node category | [Preview](https://celestiaorg.github.io/docs/pr-preview/pr-1385/nodes/celestia-app-multisig) | [Final Link](https://docs.celestia.org/nodes/celestia-app-multisig) |
| Moves Vesting doc to Node category | [Preview](https://celestiaorg.github.io/docs/pr-preview/pr-1385/nodes/celestia-app-vesting) | [Final Link](https://docs.celestia.org/nodes/celestia-app-vesting) |
| Removes links in menu to wallets, reducing bloat. Combines developer wallet integration page to just a single code block example page, with examples for each wallet as tabs. | [Preview](https://celestiaorg.github.io/docs/pr-preview/pr-1385/developers/wallets) | [Final Link](https://docs.celestia.org/developers/wallets) |
| Removes components for wallet connection. | | |
| Moves Transaction resubmission guide to Developers from Nodes category. | [Preview](https://celestiaorg.github.io/docs/pr-preview/pr-1385/developers/transaction-resubmission) | [Final Link](https://docs.celestia.org/developers/transaction-resubmission) |

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

## Summary by CodeRabbit

- **New Features**
    - Expanded documentation with new links and subcategories on topics like staking, IBC relayers, and wallet integration.
    - Introduced a broader focus on integrating Celestia chains with various wallets, including steps for adding custom chains.
- **Documentation**
    - Updated various links and texts across documentation for improved clarity and navigation.
    - Enhanced formatting in tutorials and guides for better readability.
    - Added sections for easier navigation within documents.
- **Bug Fixes**
    - Corrected an expression in the "submit-data" documentation for accuracy.
- **Refactor**
    - Removed outdated sections and content from several documents, ensuring up-to-date and focused information.

### CODE REVIEW

File: developers/wallets.md [hunk 1]
```diff
@@ -1,57 +1,26 @@
 ---
-description: How you can add Celestia network parameters to Keplr.
-next:
-  text: "Integrating Leap for developers"
-  link: "developers/leap"
+description: How you can add Celestia network parameters to wallets.
```
[the remark this answers is not in the document]
Comment (jcstein) [reply developers/wallets.md:2]: what do you suggest?

[the remark this answers is not in the document]
Comment (jcstein) [reply developers/wallets.md]: https://github.com/celestiaorg/docs/pull/1385/commits/a1560fa42407d4f31b273840d4839b1934e6ead7 thanks robbot

File: developers/wallets.md [hunk 2]
```diff
@@ -204,3 +162,57 @@
...
+- Decimals: `6`
+- Gas rate Tiny: `0.1`
+- Gas rate Low: `0.25`
+- Gas rate Average: `0.5`
+
+Now, click `Add a custom chain` and you will be able to view your Celestia
+account balance and transactions in Cosmostation wallet.
+
+Switch chains to "Mocha testnet" and you'll see that you're connected
+to Celestia's Mocha testnet!
```
[the remark this answers is not in the document]
Comment (jcstein) [reply developers/wallets.md]: https://github.com/celestiaorg/docs/pull/1385/commits/9d27b605333533c277e0c607e4993a26b1dd0e6f

**Comment by jcstein** [comment]
Screenshot 2024-02-05 at 11 49 56 AM
- [ ] double check flow and remove second specs link

File: developers/wallets.md [hunk 3]
```diff
@@ -204,3 +163,60 @@ export const ARABICA_PARAMS = {`{
...
+1. Clicking the Cosmos logo in the top corner of Leap wallet
+2. Scrolling down and clicking "Add new chain"
+
+You can
+then add the following parameters:
+
+- Chain Id: `{{ constants.arabicaChainId }}`
+- Chain Name: `Arabica devnet`
+- New RPC URL: `https://consensus-full.celestia-arabica-10.com`
+- New REST URL: `https://api.consensus.celestia-arabica-10.com`
```
Comment (Bidon15) [inline developers/wallets.md:183]:
```suggestion
- New RPC URL: `https://rpc.celestia-arabica-11.com/`
- New REST URL: `https://api.celestia-arabica-11.com`
```

**Review by Bidon15 — APPROVED** [review]
LGTM. Awesome refactor 🤴

**Pull request merged**